Add rp2040 boards

Jpe230 opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

<!--- Describe your changes in detail here. -->
<!--- Mention 'Fixed #<issue_number>' (eg 'Fixed #1234') to link to fixed issues. -->

Add commands to send images of the pinout for the Adafruit KB2040 and the Sparkfun Pro Micro RP2040

tzarc commented:
Should we perhaps make this `sparkfun2040` and `adafruit2040`? Should be easier to remember?

yanfali commented on qmkbot.js:
```diff
@@ -7,7 +7,7 @@ require('dotenv').config();
 
 // Import utils.js
-const {prefix, baseurl, msg, docsSwitch, authroles, parse, bare, firmware, toolbox, plainhelp, disclaimer, ohshitgit, git, xkcd, promicro, protonc, elitec, blackpill, bluepill, msys, coc, kbdfans, lighting, wsl, vid, vidq, automark, markdown, checkrole, sonixinvite, sonix, openrgbinvite, openrgb, vialinvite, vial, iconfrominvite, inputlanguage, standards, snip, zmk, zmkinvite, prstatus} = require ...
+const {prefix, baseurl, msg, docsSwitch, authroles, parse, bare, firmware, toolbox, plainhelp, disclaimer, ohshitgit, git, xkcd, promicro, protonc, elitec, blackpill, bluepill, sparkfunrp, adafruitkb, msys, coc, kbdfans, lighting, wsl, vid, vidq, automark, markdown, checkrole, sonixinvite, sonix, openrgbinvite, openrgb, vialinvite, vial, iconfrominvite, inputlanguage, standards, snip, zmk, zmkinv ...
```
at some point it just gets easier to stop doing this kind of import and just using utils.<something>

Jpe230 commented:
> Should we perhaps make this `sparkfun2040` and `adafruit2040`? Should be easier to remember?

The parser borks if there are numbers in the command :( that's why the commands don't use *2040

yanfali commented:
> > Should we perhaps make this `sparkfun2040` and `adafruit2040`? Should be easier to remember?
> 
> The parser borks if there are numbers in the command :( that's why the commands don't use *2040

would pico be better compromise?

Jpe230 commented:
> > > Should we perhaps make this `sparkfun2040` and `adafruit2040`? Should be easier to remember?
> > 
> > 
> > The parser borks if there are numbers in the command :( that's why the commands don't use *2040
> 
> would pico be better compromise?

Pico refers to a board from the Pi Foundation, while it uses the same MCU, it isnt part of this PR.

Jpe230 commented:
Rebased my PR and updated the REGEX to allow the use of numbers.

The commands now are `sparkfun2040` and `adafruit2040`

I tested that all current commands still works with the modified REGEX
